velox/expression/tests/SignatureTemplateBinder.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/expression/tests/SignatureTemplateBinder.h"
+
+namespace facebook::velox::test {
+
+namespace {
+
+// Return a random type among those in typeList determined by seed.
+TypePtr randomType(std::mt19937& seed) {
+  static std::vector<TypePtr> typeList{
```

Comment:
naming: kTypes or kSupportedTypes

Why are we restricting this logic to a subset of primitive types. For example, this list is missing DATE type.

Would it make sense to decouple the logic of generating random types from the rest? For example, we could use ReverseSignatureBinder class for processing the concrete return type and fill in some type variables. We could then fill the rest of the type variables with random types (or something else) and resolve argument types using the full set of bindings.